src/main/java/com/hsbc/engineering/JDBCSpecTest.java

```diff
@@ -27,21 +26,17 @@ public class JDBCSpecTest {
     void check_if_list_of_tables_can_be_extracted(Connection conn) {
         int count = 0;
 
-        try {
-            DatabaseMetaData md = conn.getMetaData();
-            ResultSet rs = md.getTables(null, null, "%", null);
-
+        try (ResultSet rs = conn.getMetaData().getTables(null, null, "%", null)) {
```

Comment:
I think this line should be like:
`
        try (conn; 
             ResultSet rs = conn.getMetaData().getTables(null, null, "%", null)) {
` in order to make sure the DB connection is closed properly.